enterprise/api/v1/views.py

```diff
@@ -1483,3 +1484,54 @@ def update(self, request, *args, **kwargs):
     )
     def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
+
+
+class EnterpriseUserLinkView(EnterpriseReadWriteModelViewSet):
+    """
+    View for
+    /enterprise/api/enterprise_user_link/{enterprise_customer_key}
+
+    Given a enterprise_customer_key, link user to the appropriate enterprise.
+
+    If the key is not found, returns 400
+    If the ket is not valid, return 422
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+
+    @action(methods=['post'], detail=False)
+    def link_user(self, request, enterprise_customer_key):
+        """
+        Post
+        Links user using enterprise_customer_key
+        """
+        try:
+            enterprise_customer_key_match = models.EnterpriseCustomerInviteKey.objects.get(uuid=enterprise_customer_key)
+
+            if not enterprise_customer_key_match.is_valid:
+                return Response(
+                    {"detail": "Enterprise customer invite key is not valid"},
+                    status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                )
+
+            enterprise_customer = enterprise_customer_key_match.enterprise_customer
+
+            _, created = models.EnterpriseCustomerUser.objects.get_or_create(
```

Comment:
following up on the discussion we were having about already linked users, we probably can't use get_or_create here with the invite key